src/chromo/common.py

```diff
@@ -543,6 +537,79 @@ def __getnewargs__(self):
         # upon unpickling, create EventData object instead of MCEvent object
         return (EventData,)
 
+    @abstractmethod
+    def _add_init_beam_info(self):
+        pass
+
+    def _get_ibeam(self):
+        if self.kin._initial_beam_data is None:
+            ibeam = self.kin._get_beam_data(self.kin)
+        else:
+            ibeam = self.kin._initial_beam_data
+
+        if ibeam.ref_frame != self._generator_frame:
+            # Boost current ref frame (ibeam.ref_frame)
+            # to generators ref frame (self._generator_frame)
+            # Use apply_boost use self.kin.frame
+            self.kin.frame = self._generator_frame
+            self.kin.apply_boost(ibeam, ibeam.ref_frame)
+            self.kin.frame = ibeam.ref_frame
+            ibeam.ref_frame = self._generator_frame
+        return ibeam
```

Comment:
This code should live in EventKinematics, not here. You can add a method `_get_beam_data` (ibeam is redundant, there is only one kind of beam) which accepts the generator frame as an argument.

Reply:
The method is moved with some modifications to EventKinematics.